substrate-node/pallets/pallet-smart-contract/service_consumer_contract_flow.md

```diff
@@ -0,0 +1,133 @@
+# Service/Consumer contracts
+
+It is now possible to create generic contract between two TFChain users (without restriction of account type) for some service and bill for it.
+
+
+## How does it work?
+
+The initial scenario is when two parties, a service provider an a consumer of the service, want to use TFChain to automatically handle the billing/payment process for an agreement they want to make.
+Multiple use cases can benefit from this feature, here are some examples of applications:
+
+//TODO give some examples of applications
```

Comment:
Lets add an example before this gets merged, or remove the "here are some examples"

Reply:
I don't have clear idea of specific applications so I removed it
If anyone has examples, feel free to add